src/render/src/pso/buffer.rs

```diff
@@ -295,33 +305,58 @@ impl<R: Resources> DataBind<R> for RawConstantBuffer {
 impl<'a, T: ToUniform> DataLink<'a> for Global<T> {
     type Init = &'a str;
     fn new() -> Self {
-        Global(None, PhantomData)
+        Global(RawGlobal::new(), PhantomData)
+    }
+    fn is_active(&self) -> bool {
+        self.0.is_active()
+    }
+    fn link_global_constant(&mut self, var: &shade::ConstVar, init: &Self::Init) ->
+                            Option<Result<(), shade::UniformValue>> {
+        self.0.link_global_constant(var, init)
```

Comment:
I believe it should be here in `link_global_constant` return type, so that you can pass it through. The old error type `shade::UniformValue` didn't make much sense and wasn't used anyway.
Btw, you should first check for the underlying compatibility of the raw stuff (`self.0.link...`) and then (literally, using `and_then`) check for compatibility.